lib/datastore/request.js

```diff
@@ -196,6 +196,75 @@ DatastoreRequest.prototype.insert = function(entities, callback) {
  *   data: {
  *     rating: '10'
  *   }
+ * }, function(err) {
+ *   console.log(key.path); // [ 'Company', 5669468231434240 ]
+ *   console.log(key.namespace); // undefined
+ * });
+ *
+ * //-
+ * // Save a single entity using a provided name instead of auto-generated ID.
+ * //
+ * // Here we are providing a key with name instead of an id. After saving, the
+ * // original Key object used to save will be updated to contain the path with
+ * // the name instead of a generated ID.
+ * //-
+ * var key = dataset.key(['Company', 'DonutShack']);
+ *
+ * dataset.save({
+ *   key: key,
+ *   data: {
+ *     name: 'DonutShack',
+ *     rating: 8
+ *   }
+ * }, function(err) {
+ *   console.log(key.path); // ['Company', 'DonutShack']
+ *   console.log(key.namespace); // undefined
+ * });
+ *
+ * //-
+ * // Save a single entity with a provided namespace. Namespaces allow for
+ * // multitenancy. To read more about this, see [here](https://goo.gl/M1LUAu).
```

Comment:
Instead of "here" can we put "the Datastore documentation on key concepts" ? (or something?)